core/src/test/java/com/squarespace/template/plugins/platform/CommerceFormattersTest.java

```diff
@@ -240,4 +240,16 @@ public void testProductRestockNotification() {
     runner.exec("f-product-restock-notification-%N.html");
   }
 
+  @Test
+  public void testSubscriptionPrice() {
+    runner.run(
+        "f-subscription-price-multiple-variants-and-multiple-pricing-options.html",
```

Comment:
So this test will go through each of the specified html files, look at the given JSON and then assert that the defined OUTPUT is the result for `{@|subscription-price}`? For my own understanding, what testing framework is this?

Reply:
Yep that is correct. The underlying framework that is used here is [TestNg](https://testng.org/#_welcome_to_testng) but I believe we own all the template parsing and conversion of formatter to string.